### ISSUE TITLE
2.4.12 Focus Appearance (Enhanced): Unobscured

### ISSUE THREAD

**Issue by JAWS-test** [w3c/wcag#1304]
> Unobscured: No part of the focus indicator is hidden by author-created content.

I consider the current wording to be worthless, as it only stipulates that the indicator must be visible, but not the focused element. This means that the requirement is lower than for level AA-SC 2.4.11 - namely precisely when the focus indicator is not a border around the element, but is only next to the element

And if it is not possible to require at 2.4.11 that the focus and the focused element must be as far as possible in the visible range, this should at least be required at 2.4.12. As already mentioned at https://github.com/w3c/wcag/issues/952 and https://github.com/w3c/wcag/issues/1145#issuecomment-638137632, the lack of visibility of focus and focused element is often not limited to author-generated content, but often also to the fact that both are simply outside the viewport

**Comment by alastc** [comment]
> I consider the current wording to be worthless, as it only stipulates that the indicator must be visible, but not the focused element.

That seems odd to me, as during the research & testing it was easy to find examples where the focus indicator would be hidden but the element was partially or completely visible. 

Have you come across examples where the indicator is present but not the element?

I'm not opposed to lining it up more closely with 2.4.11, e.g:
> **Unobscured:** No part of the item with focus is hidden by author-created content.

However, I think that is _easier_ to meet that the current wording.

**Comment by JAWS-test** [comment]
> Have you come across examples where the indicator is present but not the element?

The problem can always occur when the focus indicator is only on one side and not around the item, as in [Example 3](https://w3c.github.io/wcag/understanding/focus-appearance-minimum.html).

Therefore I think the SC should require that the focus and the focused element are fully visible, as long as the focused element is not too large for the viewport

**Comment by JAWS-test** [comment]
And for level AAA, it might even be required that form fields not only have the field in the visible area when it receives the focus, but also its label (if the field has an explicit label and is not only indirectly labelled, e.g. via a column header in a table)

**Comment by alastc** [comment]
> The problem can always occur when the focus indicator is only on one side and not around the item, as in Example 3 or Example 16.

In those examples the focus indicator is inside the element, so would be covered by the current wording.

> I think the SC should require that the focus and the focused element are fully visible, as long as the focused element is not too large for the viewport

Ok, so this is essentially a duplicate of #1271, saying that the AA version should be 'fully' visible rather than allow for particial obscurement.

**Comment by alastc** [comment]
Hi everyone,

As per #1271, the group agreed this response to this issue: 
<hr>
There is significant evidence that it is not always possible for authors to meet a strict 'fully unobscured' requirement. Therefore that requirement has been kept to the AAA version of Focus Appearance.
